### PR TITLE
add spring-boot generator for cors customization

### PR DESCRIPTION
**Pull request by mshima**

### CODE REVIEW

**Comment by mshima** [comment]
Cors urls needs adjusts

File: generators/spring-boot/templates/src/main/resources/config/application-e2e-cors.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+jhipster:
+  cors:
+    allowed-origins: "http://localhost:8100,capacitor://localhost,http://localhost"
```
Comment (mraible) [inline generators/spring-boot/templates/src/main/resources/config/application-e2e-cors.yml:3]:
```suggestion
    allowed-origins: "http://localhost:8081,http://localhost"
```

File: generators/spring-boot/generator.mjs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  get [BaseApplicationGenerator.POST_WRITING]() {
+    return this.asPostWritingTaskGroup({
+      async postWritingTemplateTask() {
+        this.editFile('src/main/docker/app.yml', content =>
+          content.replace('SPRING_PROFILES_ACTIVE=prod,api-docs', 'SPRING_PROFILES_ACTIVE=prod,api-docs,e2e-cors'),
+        );
+        this.editFile('src/main/resources/config/application.yml', content =>
+          content.replace(/allowed-origins: (['"])(.*)['"]/, 'allowed-origins: $1$2,capacitor://localhost,http://localhost$1'),
```
Comment (mraible) [inline generators/spring-boot/generator.mjs:32]:
```suggestion
          content.replace(/allowed-origins: (['"])(.*)['"]/, 'allowed-origins: $1$2,http://localhost$1'),
```

File: generators/spring-boot/generator.mjs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        );
+        this.editFile('src/main/resources/config/application-dev.yml', content =>
+          content.replace(/allowed-origins: (['"])(.*)['"]/, 'allowed-origins: $1$2,capacitor://localhost,http://localhost$1'),
```
Comment (mraible) [inline generators/spring-boot/generator.mjs:35]:
```suggestion
          content.replace(/allowed-origins: (['"])(.*)['"]/, 'allowed-origins: $1$2,http://localhost$1'),
```

File: generators/spring-boot/generator.mjs (hunk 2 above)
Comment (mraible) [reply generators/spring-boot/generator.mjs:32]: Should `http://localhost:8081` be specified here too? It's needed for web.

Comment (mshima) [reply generators/spring-boot/generator.mjs:32]: I don't know.

File: generators/spring-boot/generator.mjs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  get [BaseApplicationGenerator.POST_WRITING]() {
+    return this.asPostWritingTaskGroup({
+      async postWritingTemplateTask() {
+        this.editFile('src/main/docker/app.yml', content =>
+          content.replace('SPRING_PROFILES_ACTIVE=prod,api-docs', 'SPRING_PROFILES_ACTIVE=prod,api-docs,e2e-cors'),
+        );
+        this.editFile('src/main/resources/config/application.yml', content =>
+          content.replace(/allowed-origins: (['"])(.*)['"]/, 'allowed-origins: $1$2,http://localhost$1'),
```
Comment (mraible) [inline generators/spring-boot/generator.mjs:32]:
```suggestion
          content.replace(/allowed-origins: (['"])(.*)['"]/, 'allowed-origins: $1$2,http://localhost:8081,http://localhost$1'),
```

File: generators/spring-boot/generator.mjs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        );
+        this.editFile('src/main/resources/config/application-dev.yml', content =>
+          content.replace(/allowed-origins: (['"])(.*)['"]/, 'allowed-origins: $1$2,http://localhost$1'),
```
Comment (mraible) [inline generators/spring-boot/generator.mjs:35]:
```suggestion
          content.replace(/allowed-origins: (['"])(.*)['"]/, 'allowed-origins: $1$2,http://localhost:8081,http://localhost$1'),
```

**Review by mraible — APPROVED** [review]

**Pull request merged**